hparam_search.py

```diff
@@ -0,0 +1,105 @@
+import torch
+import numpy as np
+from skopt import BayesSearchCV
+from skopt.space import Integer, Real
+from sklearn.base import BaseEstimator, RegressorMixin
+
+from train import QModelTrainRun
+from model import CheckersQModel
+
+class CheckersQModelWrapper(BaseEstimator, RegressorMixin):
+    def __init__(self, num_hidden_layers=0, hidden_size=0, learning_rate=0.01,
+                 self_play_games_per_iter=25, weight_decay=1e-4):
+        self.num_hidden_layers = num_hidden_layers
+        self.hidden_size = hidden_size
+        self.learning_rate = learning_rate
+        self.self_play_games_per_iter = self_play_games_per_iter
+        self.weight_decay = weight_decay
+
+        # Initialize model here
+        self.model = None
+
+        # Initialize train_hist to an empty list
+        self.train_hist = []
+
+    def fit(self, X, y=None, **kwargs):
+        # Update the model's parameters
+        self.model = CheckersQModel(
+            num_hidden_layers=self.num_hidden_layers,
+            hidden_size=self.hidden_size
+        )
+
+        # Set other parameters as needed
+        self.optimzer = torch.optim.SGD(self.model.parameters(),
+                                        lr=self.learning_rate,
+                                        weight_decay=self.weight_decay)
+        self.train_run = QModelTrainRun(self.model, self.optimzer)
+
+        # Train the model and get the metric to optimize (e.g., average win rate)
+        self.train_hist = self.train_run.train(selfplay_games_p_i=self.self_play_games_per_iter)
+        return self
+
+    def score(self, X, y=None, **kwargs):
+        num_evaluation_games = 100
+        wins, draws = self.train_run.evaluate_strength(num_evaluation_games=num_evaluation_games)
+        return wins / num_evaluation_games
```

Comment:
I removed the evaluation games that are played every iteration of the train loop, since they took a lot unnecessary time. Instead there's now a new function (evaluate_strength) to play these games vs random moves, after a training run is completed.
So I changed this here to work with that

Reply:
The score method now returns the winrate. Idk how skopt works, does it try to minimize the score or maximize it? If it minimizes it, we should add the minus here (`return -wins / num_evaluation_games`)